Signal/src/ViewControllers/MessageDetailViewController.swift

```diff
@@ -212,7 +212,16 @@ class MessageDetailViewController: OWSViewController, MediaGalleryDataSourceDele
                     groupRows.append(divider)
                 }
 
-                for recipientId in thread.recipientIdentifiers {
+                var messageRecipientIds: [String]
+                if let intendedRecipientIds = outgoingMessage.intendedRecipientIds {
+                    // New messages capture the list of intended recipients.
+                    messageRecipientIds = Array(intendedRecipientIds)
+                } else {
+                    // For legacy messages, just use the list of members of the thread.
+                    messageRecipientIds = thread.recipientIdentifiers
+                }
```

Comment:
First fix: in message details view, only show the "intended recipients" which are captured the first time we try to send the message.